content/en/blog/_posts/2025-07-28-kubernetes-v1.34-sneak-peek.md

```diff
@@ -0,0 +1,132 @@
+
+Since the v1.30 release, DRA is based around RC concept of claiming devices using _structured parameters_ that are opaque to the core of Kubernetes.
+The relevant enhancement proposal, [KEP-4381](https://kep.k8s.io/4381), took inspiration from dynamic provisioning for storage volumes.
+DRA with structured parameters relies on a set of supporting API kinds: ResourceClaim, DeviceClass, ResourceClaimTemplate, and ResourceSlice API types under `resource.k8s.io`, while extending the `.spec` for Pods with a new `resourceClaims` field.
+The core of DRA is targeting graduation to stable in Kubernetes v1.34.
+
+
+With DRA, device drivers and cluster admins define device classes that are available for use. Workloads can claim devices from a device class within device requests. Kubernetes allocates matching devices to specific claims and places the corresponding Pods on nodes that can access the allocated devices. This framework provides flexible device filtering using CEL, centralized device categorization, and simplified Pod requests, among other benefits.
+
+Once this feature has graduated, the `resource.k8s.io/v1` APIs will be available by default.
+
+### ServiceAccount tokens for image pull authentication
+
+The existing [ServiceAccount](/docs/concepts/security/service-accounts/) token integration for `kubelet` credential providers is likely to reach beta in Kubernetes v1.34, and is likely to be enabled by default. You'll then be able to have the `kubelet` use these tokens when pulling container images from registries that require authentication.
+
+That support already exists as alpha, and is tracked as part of [KEP-4412](https://kep.k8s.io/4412).
+
+The existing alpha integration allows the `kubelet` to use short-lived, automatically rotated ServiceAccount tokens (that follow OIDC-compliant semantics) to authenticate to a container image registry. Each token is scoped to one associated Pod; the overall mechanism replaces the need for long-lived image pull Secrets.
+
+Adopting this new approach reduces security risks, supports workload-level identity, and helps to cut operational overhead. It brings image pull authentication closer to modern, identity-aware good practice.
+
+### Pod replacement policy for Deployments
+
+After a change to a [Deployment](/docs/concepts/workloads/controllers/deployment/), terminating pods may stay up for a considerable amount of time and may consume additional resources.
+As part of [KEP-3973](https://kep.k8s.io/3973), the `.spec.podReplacementPolicy` field will be introduced (as alpha) for Deployments.
+
+If your cluster has the feature enabled, you'll be able to select one of two policies:
+
+`TerminationStarted`
+: Creates new pods as soon as old ones start terminating, resulting in faster rollouts and temporary higher resource consumption.
+
+`TerminationComplete`
+: Waits until old pods fully terminate before creating new ones, resulting in slower rollouts and a more controlled resource consumption.
```

Comment:
If we were to reword the above for `TerminationStarted`, it would be good to keep this one consistent as well.

```suggestion
: Waits until old pods fully terminate before creating new ones, slowing down rollouts but ensuring controlled resource consumption.
```

Or
```suggestion
: Waits until old pods fully terminate before creating new ones, resulting in slower rollouts but ensuring controlled resource consumption.
```